common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -46,6 +47,42 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
    * @return true if the arrays are equal, false otherwise
    */
   public static boolean arrayEquals(
+      MemoryBlock leftBase, long leftOffset, MemoryBlock rightBase, long rightOffset, final long length) {
+    int i = 0;
+
+    // check if stars align and we can get both offsets to be aligned
```

Comment:
stars align...? :)... starts?

Reply:
lol :) If you scroll below, this is copy-paste from the L89 below .. so not introduced by this PR. I feel this might not be a typo and the author legitimately meant that.

Reply:
fixed :)

Reply:
⭐️ ⭐️ ⭐️